big_tests/tests/inbox_SUITE.erl

```diff
@@ -299,21 +314,62 @@ end_per_testcase(CaseName, Config) ->
-        escalus:send(Alice, inbox_helper:get_inbox_form_stanza()),
-        ResIQ = escalus:wait_for_stanza(Alice),
-        InboxNS = inbox_helper:inbox_ns(),
-        #{ field_count := 5 } = Form = parse_form_iq(ResIQ),
-        #{ <<"FORM_TYPE">> := #{ type := <<"hidden">>,
-                                 value := InboxNS } } = Form,
-        #{ <<"start">> := #{ type := <<"text-single">> } } = Form,
-        #{ <<"end">> := #{ type := <<"text-single">> } } = Form,
-        #{ <<"order">> := #{ type := <<"list-single">>,
-                             value := <<"desc">>,
-                             options := OrderOptions } } = Form,
-        [<<"asc">>, <<"desc">>] = lists:sort(OrderOptions),
-        #{ <<"hidden_read">> := #{ type := <<"text-single">> } } = Form
-      end).
+  escalus:story(Config, [{alice, 1}], fun(Alice) ->
+      escalus:send(Alice, inbox_helper:get_inbox_form_stanza()),
+      ResIQ = escalus:wait_for_stanza(Alice),
+      InboxNS = inbox_helper:inbox_ns(),
+      #{ field_count := 5 } = Form = parse_form_iq(ResIQ),
+      #{ <<"FORM_TYPE">> := #{ type := <<"hidden">>,
+                               value := InboxNS } } = Form,
+      #{ <<"start">> := #{ type := <<"text-single">> } } = Form,
+      #{ <<"end">> := #{ type := <<"text-single">> } } = Form,
+      #{ <<"order">> := #{ type := <<"list-single">>,
+                           value := <<"desc">>,
+                           options := OrderOptions } } = Form,
+      [<<"asc">>, <<"desc">>] = lists:sort(OrderOptions),
+      #{ <<"hidden_read">> := #{ type := <<"text-single">> } } = Form
+    end).
+
+returns_error_when_bad_form_field_order_sent(Config) ->
+  escalus:story(Config, [{alice, 1}], fun(Alice) ->
+      assert_invalid_inbox_form_value_error(Alice, <<"order">>, <<"invalid">>)
+    end).
+
+returns_error_when_bad_form_field_start_sent(Config) ->
+  escalus:story(Config, [{alice, 1}], fun(Alice) ->
+      assert_invalid_inbox_form_value_error(Alice, <<"start">>, <<"invalid">>)
+    end).
+
+returns_error_when_bad_form_field_end_sent(Config) ->
+  escalus:story(Config, [{alice, 1}], fun(Alice) ->
+      assert_invalid_inbox_form_value_error(Alice, <<"end">>, <<"invalid">>)
+    end).
+
+returns_error_when_bad_form_field_hidden_read_sent(Config) ->
+  escalus:story(Config, [{alice, 1}], fun(Alice) ->
+      assert_invalid_inbox_form_value_error(Alice, <<"hidden_read">>, <<"invalid">>)
+    end).
+
+returns_error_when_first_bad_form_field_encountered(Config) ->
+  escalus:story(Config, [{alice, 1}], fun(Alice) ->
+      Stanza = inbox_helper:get_inbox_stanza( #{ <<"start">> => <<"invalid">>,
+                                                 <<"end">> => <<"invalid">>}, false),
+      escalus:send(Alice, Stanza),
+      [ResIQ] = escalus:wait_for_stanzas(Alice, 1),
+      escalus_pred:is_iq_error(ResIQ),
+      <<"Invalid inbox form field value, field=end, value=invalid">> = inbox_helper:get_error_message(ResIQ)
```

Comment:
It might be better to check if the error message contains `field=end` & `value=invalid` - this way most changes in format or contents of the error message won't trigger a test failure.